### PR TITLE
[FLINK-23645][Table SQL / Client] Fix SqlClient doesn't response correctly to Signal.INT

### PR DESCRIPTION
**Pull request by saddays**
## What is the purpose of the change

*Fix SqlClient doesn't response correctly to Signal.INT*

## Brief change log

*Sqlclient would interupt the Executor when receive Signal.INT before ResultView opening*

## Verifying this change

*Added test case in **org.apache.flink.table.client.cli.CliClientTest.testCancelExecutionInteractiveMode***

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by saddays** [comment]
hi @wuchong , do you have time review this pr ?

**Comment by wuchong** [comment]
cc @fsk119

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java [hunk 1]
```diff
@@ -389,51 +395,58 @@ private void validate(Operation operation, ExecutionMode executionMode) {
...
-        } else if (operation instanceof ShowJarsOperation) {
-            // SHOW JARS
-            callShowJars();
-        } else if (operation instanceof ShowCreateTableOperation) {
-            // SHOW CREATE TABLE
-            callShowCreateTable((ShowCreateTableOperation) operation);
-        } else {
-            // fallback to default implementation
-            executeOperation(operation);
+        final Thread thread = Thread.currentThread();
```
Comment (fsk119) [inline flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java:398]: How about move the logic about register  handler to the `executeStatement` part. We may add more commands in the future and the block will be larger and lager.

```
           operation.ifPresent(
                    op -> {
                        final Thread thread = Thread.currentThread();
                        final Terminal.SignalHandler previousHandler =
                                terminal.handle(
                                        Terminal.Signal.INT, (signal) -> thread.interrupt());
                        try {
                            callOperation(op, executionMode);
                        } finally {
                            if (previousHandler != null) {
                                terminal.handle(Terminal.Signal.INT, previousHandler);
                            }
                        }
                    });
```

**Review by fsk119 — COMMENTED** [review]
Thanks for your great work. I left some comments.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java [hunk 2]
```diff
@@ -324,7 +325,12 @@ private boolean executeStatement(String statement, ExecutionMode executionMode)
             final Optional<Operation> operation = parseCommand(statement);
             operation.ifPresent(op -> callOperation(op, executionMode));
         } catch (SqlExecutionException e) {
-            printExecutionException(e);
+            Throwable t = ExceptionUtils.getRootCause(e);
+            if (t instanceof InterruptedException) {
+                printExecutionException(t);
+            } else {
+                printExecutionException(e);
+            }
```
[the remark this answers is not in the document]
Comment (saddays) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java:333]: you are right, i will fix it.

File: flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java (hunk 1 above)
Comment (saddays) [reply flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java:398]: Seems much more reasonable,  fix it later.

**Review by fsk119 — APPROVED** [review]
LGTM

**Pull request merged**